cli/registry/client/client.go

```diff
@@ -67,9 +69,13 @@ func (c *client) MountBlob(ctx context.Context, sourceRef reference.Canonical, t
 	default:
 		return errors.Wrapf(err, "failed to mount blob %s to %s", sourceRef, targetRef)
 	}
-
-	lu.Cancel(ctx)
-	return errors.Errorf("failed to mount blob %s to %s", sourceRef, targetRef)
+	if lu != nil {
```

Comment:
I don't think `lu` can ever be `nil` here, since we've already checked that `err == nil` above.
https://github.com/docker/distribution/blob/master/registry/client/repository.go#L704-L756

Did you encounter a case where it was nil?

I don't understand why this case is also a success case. According to the docs `StatusAccepted` means it failed: https://docs.docker.com/registry/spec/api/#cross-repository-blob-mount

> If a mount fails due to invalid repository or digest arguments, the registry will fall back to the standard upload behavior and return a 202 Accepted with the upload URL